assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -464,30 +472,25 @@ const baseActions = {
 		},
 		function* ( slugs ) {
 			const { dispatch, select } = yield Data.commonActions.getRegistry();
+			const { response } =
+				yield fetchRecoverModulesStore.actions.fetchRecoverModules(
+					slugs
+				);
+			const { success } = response;
 
-			const recoveredModules = [];
-
-			const errors = [];
-
-			for ( const slug of slugs ) {
-				const { response, error } =
-					yield fetchRecoverModuleStore.actions.fetchRecoverModule(
-						slug
-					);
+			const successfulRecoveries = Object.keys( success ).filter(
+				( slug ) => !! success[ slug ]
+			);
 
-				if ( response?.ownerID ) {
-					const storeName =
-						select( CORE_MODULES ).getModuleStoreName( slug );
-					// Reload the module's settings from the server.
-					yield dispatch( storeName ).fetchGetSettings();
+			for ( const slug of successfulRecoveries ) {
+				const storeName =
+					select( CORE_MODULES ).getModuleStoreName( slug );
 
-					recoveredModules.push( slug );
-				} else {
-					errors.push( [ slug, error ] );
-				}
+				// Reload the module's settings from the server.
+				yield dispatch( storeName ).fetchGetSettings();
```

Comment:
This `yield` is unnecessary because it's dispatching directly. If the intention is to await the action, we'd need to use `yield Data.commonActions.await()` around it since the dispatched action will return a promise. In this case I think we can simply remove it as we don't depend on the response?

Ideally, it might be good to use the await action with a `Promise.all` so these can be fetched in parallel.

Reply:
Thank you, @aaemnnosttv! I've opened #6117 to address this.